src/proposals/Proposals.jl

```diff
@@ -138,5 +139,97 @@ function (prop::RWalk)(rng::AbstractRNG,
     return u, v, logl, ncall
 end
 
+"""
+    Proposals.RStagger(;ratio=0.5, walks=25, scale=1)
+
+Propose a new live point by random staggering away from an existing live point. 
+This differs from the random walk proposal in that the step size here is exponentially adjusted
+to reach a target acceptance rate _during_ each proposal, in addition to _between_
+proposals.
+
+`ratio` is the target acceptance ratio, `walks` is the minimum number of steps to take, and `scale` is the proposal distribution scale, which will update _between_ proposals.
+"""
+
+@with_kw mutable struct RStagger <: AbstractProposal
+    ratio = 0.5
+    walks = 25
+    scale = 1.0
+end
 
+function (prop::RStagger)(rng::AbstractRNG,
+    point::AbstractVector,
+    logl_star,
+    bounds::AbstractBoundingSpace,
+    loglike,
+    prior_transform;
+    kwargs...)
+    #setup
+    n = length(point)
+    scale_init = prop.scale
+    accept = reject = fail = nfail = nc = ncall = 0
+    stagger = 1
+    local drhat, dr, du, u_prop, logl_prop, u, v, logl
```

Comment:
Remove the unnecessary variables
```suggestion
    local du, u_prop, logl_prop, u, v, logl
```